testing/tests/ui/crate_identifier.rs

```diff
@@ -0,0 +1,15 @@
+use askama::Template;
+
+#[derive(Template)]
+#[template(ext = "html", source = "{{ crate }}")]
+struct Example;
+
+#[derive(Template)]
+#[template(ext = "html", source = "{% if crate == 12 %}{% endif %}")]
+struct Example2;
+
+#[derive(Template)]
+#[template(ext = "html", source = "{% match crate %}{% endmatch %}")]
+struct Example3;
```

Comment:
Can you add tests for `{% let crate %}`, `{% let crate = 12 %}` and `{{ self.crate }}`?

Reply:
About `self.crate` (or more precisely, any `.crate`): in this case, it means the user is actually doing a typo because the rust compiler doesn't allow to have a field named `crate`. So I suggest that we don't handle this case ourselves and hope that in the future, we'll have better interconnection between template source and generated rust code errors because otherwise we'll need to greatly improve our name checker (also in the `identifier` function which handles paths like `crate::whatever` where `crate` is valid in this position).

Reply:
We have to handle this case. No [`proc_macro::Ident`](https://doc.rust-lang.org/stable/proc_macro/struct.Ident.html) can be called "crate", "self", … so the parsing of our generated code will fail.